server/service/service_campaigns.go

```diff
@@ -97,7 +97,14 @@ func (svc Service) StreamCampaignResults(ctx context.Context, conn *websocket.Co
 	// Setting the status to completed stops the query from being sent to
 	// targets. If this fails, there is a background job that will clean up
 	// this campaign.
-	defer svc.CompleteCampaign(ctx, campaign) //nolint:errcheck
+	defer func() {
+		// We do not want to use the outer `ctx` because we want to make sure
+		// to cleanup the campaign.
+		ctx := context.Background()
```

Comment:
Hmm just took a look at how `StreamCampaignResults` is called, are you sure this will change anything? It doesn't look like the `ctx` it receives is cancellable anyway: it gets created [here](https://github.com/fleetdm/fleet/blob/2d73b99bb0ddaba274800017299ece31d41b463b/server/service/endpoint_campaigns.go#L84) and passed without change to `StreamCampaignResults` afterwards.

Reply:
Yes. This change is just for future devs like me that will copy+paste (future live query APIs).